### ISSUE TITLE
Canvas element renders with border in PDF

### ISSUE THREAD

**Issue by DanielCeregatti** [wkhtmltopdf/wkhtmltopdf#2146]
I'm using wkhtmltopdf 0.12.2 on 64 bit Debian Wheezy. I've enabled the Xvfb server and have the DISPLAY environment variable set. The problem I'm encountering is that a jqplot (http://www.jqplot.com/) pie chart that uses canvas elements renders with a border in the PDF despite the fact that no such border exists when viewed in chrome. No such border appears when printing my test page (http://ceregatti.org/jqplot) to PDF directly within chrome either. I'm invoking wkhtmltopdf with the following command line:

wkhtmltopdf --enable-javascript --load-error-handling abort --load-media-error-handling abort --debug-javascript --load-media-error-handling ignore  --javascript-delay "5000" --orientation Portrait http://ceregatti.org/jqplot test.pdf

It seems the border coincides with many of the canvas elements on my test page. I've tried setting various CSS properties such as "border: none;" and "border: 0px", all with no effect.

I'm hoping someone can shed some light on this.

Thanks,

Daniel

**Comment by ashkulz** [comment]
Looks like it is related to #1964.

**Comment by DanielCeregatti** [comment]
In my cases the pie charts are visible in the PDF, but they are surrounded by a border. Here's an example of what it looks like in a browser:
![selection_011](https://cloud.githubusercontent.com/assets/384719/5747430/d6fff8e4-9bec-11e4-9d5c-2a5d162d7378.png)
And here's what it looks like in the PDF:
![selection_010](https://cloud.githubusercontent.com/assets/384719/5747436/db10d5f2-9bec-11e4-9586-60aeee2a8557.png)
I've actually made some progress on this. I've used javascript to remove all the elements with the .jqplot-grid-canvas class, and no more borders. Here's my test URL:

http://ceregatti.org/jqplot/test.html

Note that it removes that node with a 2 second javascript timer. I've tested this by using a 5 second delay in wkhtmltopdf.

**Comment by ashkulz** [comment]
Are you able to get a better display with `0.13 alpha`?

**Comment by DanielCeregatti** [comment]
I don't see one for 64 bit Debian Wheezy. Once I see a package for it, I'll try it.

**Comment by Romiste** [comment]
Hi guyz
I have the same issue on the same elements (javascript)
0.13 alpha is not available on Jessie environement.
Did you find how to correct this bug ?
The display is a little bit nasty...

**Comment by Eyewash01** [comment]
I was having this problem with amCharts, whereby there were seemingly random borders around lots of things when rendered in WK.  On closer inspection, it appears to be that it isn't taking notice of "stroke-opacity" attributes, and is just rendering everything as opacity "1".

**Comment by anhvupham** [comment]
@Eyewash01 How did you make it work with Amchart ? I can't even have the graph displayed in pdf ! Can you please explain how you did it ?

**Comment by anhvupham** [comment]
Never mind, I've got it worked. But still have problem with the border ! :(

**Comment by Eyewash01** [comment]
Yeah they didn't work for me initially - think I had to specify an explicit width and height or something like that.

But yeah never solved the problem of the lines.  Kind of did a half way house where I ran through all elements with a stroke on and removed it, but obviously far from ideal.

Instead, I pre-rendered the charts as an image, then just put the images into the PDF (although that has its own problems with DPI etc - no real solution!)

**Comment by anhvupham** [comment]
Yeah, I'm currently have to remove the path stroke attribute, that's not an ideal at all but it works for now. Hopefully in new version it will be fixed.
